module/zfs/dmu.c

```diff
@@ -1993,6 +1993,9 @@ dmu_write_policy(objset_t *os, dnode_t *dn, int level, int wp, zio_prop_t *zp)
 			checksum = ZIO_CHECKSUM_FLETCHER_4;
 
 		if (os->os_redundant_metadata == ZFS_REDUNDANT_METADATA_ALL ||
+		    (os->os_redundant_metadata ==
+		    ZFS_REDUNDANT_METADATA_SOME &&
+		    DMU_OT_IS_CRITICAL(type)) ||
```

Comment:
It's functionally equivalent but let's convert this to a switch statement to make the policy more readable.  Something like this:

```c
                switch (os->os_redundant_metadata) {
                case ZFS_REDUNDANT_METADATA_ALL:
                        copies++;
                        break;
                case ZFS_REDUNDANT_METADATA_MOST:
                        if (level >= zfs_redundant_metadata_most_ditto_level ||
                            DMU_OT_IS_METADATA(type) || (wp & WP_SPILL))
                                copies++;
                        break;
                case ZFS_REDUNDANT_METADATA_SOME:
                        if (DMU_OT_IS_CRITICAL(type))
                                copies++;
                        break;
                case ZFS_REDUNDANT_METADATA_NONE:
                        break;
                }
```

Reply:
yep, changed it. Thanks for the suggestion.